analyzer/src/main/kotlin/managers/Pip.kt

```diff
@@ -424,14 +424,17 @@ class Pip(
 
         // Example license classifier:
         // "License :: OSI Approved :: GNU Library or Lesser General Public License (LGPL)"
-        pkgInfo["classifiers"]?.mapNotNullTo(declaredLicenses) { classifier ->
-            val components = classifier.textValue().split(" :: ")
-            components.takeIf { it.first() == "License" }?.last()?.takeUnless { it == "OSI Approved" }
-        }
+        pkgInfo["classifiers"]?.mapNotNullTo(declaredLicenses) { getLicenseFromClassifier(it.textValue()) }
 
         return declaredLicenses
     }
 
+    private fun getLicenseFromClassifier(classifier: String): String? {
+        val components = classifier.split(" :: ")
+
+        return components.takeIf { it.first() == "License" }?.last()?.takeUnless { it == "OSI Approved" }
```

Comment:
I think you could as well eliminate the `components` variable and make this an expression function if you want.